File: go/docmost/services/shares_api.go
package services

import (
	"context"

	"github.com/ggpo1/docmost-sdk/go/internal/httpclient"
	"github.com/ggpo1/docmost-sdk/go/docmost/models"
)

// SharesAPI implements Shares API.
type SharesAPI struct {
	http *httpclient.HTTPClient
}

func NewSharesAPI(http *httpclient.HTTPClient) *SharesAPI {
	return &SharesAPI{http: http}
}

func (a *SharesAPI) GetShares(ctx context.Context, request *models.PaginationOptions) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "shares", request)
}

func (a *SharesAPI) GetSharedPageInfo(ctx context.Context, request *models.ShareInfoDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "shares/page-info", request)
}

func (a *SharesAPI) GetShare(ctx context.Context, request *models.ShareIdDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "shares/info", request)
}

func (a *SharesAPI) GetShareForPage(ctx context.Context, request *models.SharePageIdDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "shares/for-page", request)
}

func (a *SharesAPI) CreateShare(ctx context.Context, request *models.CreateShareDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "shares/create", request)
}

func (a *SharesAPI) UpdateShare(ctx context.Context, request *models.UpdateShareDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "shares/update", request)
}

func (a *SharesAPI) DeleteShare(ctx context.Context, request *models.ShareIdDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "shares/delete", request)
}

func (a *SharesAPI) GetSharePageTree(ctx context.Context, request *models.ShareIdDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "shares/tree", request)
}

File: python/docmost/services/shares_api.py
"""Generated API client."""
from __future__ import annotations

from typing import Any, Optional

import httpx

from docmost.http_client import DocmostHttpClient
from docmost.types import ApiResponse
from docmost.models import CreateShareDto, PaginationOptions, ShareIdDto, ShareInfoDto, SharePageIdDto, UpdateShareDto


class SharesApi:
    """Shares API."""

    def __init__(self, http: DocmostHttpClient) -> None:
        self._http = http

    def get_shares(self, request: PaginationOptions) -> ApiResponse[Any]:
        return self._http.post("shares", body=request.model_dump(by_alias=True, exclude_none=True))

    def get_shared_page_info(self, request: ShareInfoDto) -> ApiResponse[Any]:
        return self._http.post("shares/page-info", body=request.model_dump(by_alias=True, exclude_none=True))

    def get_share(self, request: ShareIdDto) -> ApiResponse[Any]:
        return self._http.post("shares/info", body=request.model_dump(by_alias=True, exclude_none=True))

    def get_share_for_page(self, request: SharePageIdDto) -> ApiResponse[Any]:
        return self._http.post("shares/for-page", body=request.model_dump(by_alias=True, exclude_none=True))

    def create_share(self, request: CreateShareDto) -> ApiResponse[Any]:
        return self._http.post("shares/create", body=request.model_dump(by_alias=True, exclude_none=True))

    def update_share(self, request: UpdateShareDto) -> ApiResponse[Any]:
        return self._http.post("shares/update", body=request.model_dump(by_alias=True, exclude_none=True))

    def delete_share(self, request: ShareIdDto) -> ApiResponse[Any]:
        return self._http.post("shares/delete", body=request.model_dump(by_alias=True, exclude_none=True))

    def get_share_page_tree(self, request: ShareIdDto) -> ApiResponse[Any]:
        return self._http.post("shares/tree", body=request.model_dump(by_alias=True, exclude_none=True))

